src/merge_iterator.rs

```diff
@@ -150,11 +189,18 @@ impl<T: KeyValueIterator> PartialOrd<Self> for MergeIteratorHeapEntry<T> {
 
 impl<T: KeyValueIterator> Ord for MergeIteratorHeapEntry<T> {
     fn cmp(&self, other: &Self) -> Ordering {
-        match self.next_kv.key.cmp(&other.next_kv.key) {
-            Ordering::Less => Ordering::Less,
-            Ordering::Greater => Ordering::Greater,
-            Ordering::Equal => self.index.cmp(&other.index),
-        }
+        // the row entry with higher index, should always has higher seqnum
```

Comment:
nit: it's actually the lower indexes that should have higher seqnum as they appear first in the list